pkg/services/vmoperator/resource_policy.go

```diff
@@ -37,15 +40,31 @@ type RPService struct {
 // ReconcileResourcePolicy ensures that a VirtualMachineSetResourcePolicy exists for the cluster
 // Returns the name of a policy if it exists, otherwise returns an error.
 func (s *RPService) ReconcileResourcePolicy(ctx context.Context, clusterCtx *vmware.ClusterContext) (string, error) {
-	resourcePolicy, err := s.getVirtualMachineSetResourcePolicy(ctx, clusterCtx)
+	clusterModuleGroups, err := getClusterModuleGroups(ctx, s.Client, clusterCtx.Cluster)
+	if err != nil {
+		return "", err
+	}
+
+	resourcePolicy, err := getVirtualMachineSetResourcePolicy(ctx, s.Client, clusterCtx.Cluster)
 	if err != nil {
 		if !apierrors.IsNotFound(err) {
 			return "", errors.Errorf("unexpected error in getting the Resource policy: %+v", err)
 		}
-		resourcePolicy, err = s.createVirtualMachineSetResourcePolicy(ctx, clusterCtx)
+		resourcePolicy, err = s.createVirtualMachineSetResourcePolicy(ctx, clusterCtx, clusterModuleGroups)
 		if err != nil {
 			return "", errors.Errorf("failed to create Resource Policy: %+v", err)
 		}
+		return resourcePolicy.Name, nil
+	}
+
+	// Ensure .spec.clusterModuleGroups is up to date.
+	helper, err := patch.NewHelper(resourcePolicy, s.Client)
```

Comment:
Note: I prefer this approach (using the patch helper), but let's consider also using createOrUpdate for consistency with the rest of the codebase